piet-web/examples/basic/build.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+set -ex
+
+# Build the `hello_world.wasm` file using Cargo/rustc
+cargo build --target wasm32-unknown-unknown
+
+# Run the `wasm-bindgen` CLI tool to postprocess the wasm file emitted by the
+# Rust compiler to emit the JS support glue that's necessary
+wasm-bindgen ../../../target/wasm32-unknown-unknown/debug/piet_web_example.wasm --out-dir basic-web-static/dist
+
+# Finally, package everything up using Webpack and start a server so we can
+# browse the result
+npm install --prefix basic-web-static
```

Comment:
This also has a paper-cut error: works fine if I cd to basic-web-static and do `npm install` and serve from there, but not from the dir as shown in the batch file. But then it works!

Reply:
Hmm... the --prefix option should take care of that. I don't need to cd into that directory on my machine. Maybe different versions of npm are causing this? 

I can change the script to cd into that directory instead. Do you think that will be a more robust solution? (ah... the joys of bash scripting ❤️)

Reply:
The world of npm is foreign and strange to me, though I have played with it some. Looking at the [docs](https://docs.npmjs.com/misc/config#prefix), it looks like prefix sets the path where the artifacts get stored, but not where the package.json is read from. Did you try this from a fresh clone? (my inexperienced experience is that npm tends to litter a lot of files in your working directory)

I'm fine with it cd'ing into the directory. In general I would want to follow "best practices" but have no idea what those are. Not to be too nitpicky, but I think this is kinda important because a *lot* of people start projects by copying a hello example.

Reply:
No worries on being nitpicky. I 100% that this should be painless for people. 

What you said about `--prefix` is true for global items (installed with `-g`) but for local cli commands this tells npm in which directory to run. From the docs: "If set on the command line, then it forces non-global commands to run in the specified folder."

I tried from a fresh clone and it worked fine for me. 

If you run `npm install --prefix basic-web-static` from that directory (the $PROJECT_ROOT/piet-web/examples/basic` directory), it warns that you don't have a `package.json`?

I try to avoid using cd in my bash scripts because I've ran into weird issues before, but if that's what's needed to get it working, we can do it.

Reply:
```
raphl@DESKTOP-6JFLCBE MINGW64 ~/github/piet/piet-web/examples/basic (rylev-wasm-bindgen)
$ npm -v
6.4.1
$ npm install --prefix basic-web-static
npm ERR! code ENOLOCAL
npm ERR! Could not install from "" as it does not contain a package.json file.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\raphl\AppData\Roaming\npm-cache\_logs\2019-01-07T17_39_10_724Z-debug.log
```

I'm not sure what's going on here, and totally understand wanting to minimize cd'ing.

Reply:
Looks like a bug in the windows version of npm: https://npm.community/t/npm-prefix-flag-does-not-work-as-expected-on-windows/3147

I think moving to just using `cd` is the best choice.

Reply:
Sounds good - I was just about to try it on my Mac, but seems like you've isolated it. That bug is scary, being auto-closed even though it seems valid and with the potential to affect lots of people.

Reply:
That's unfortunately how the web world works. It's pretty fast and loose. The latest commit _should_ fix it (though I just ran into another way things could break that the readme now references).